packages/teleport/src/Apps/AddApp/Manually/Manually.tsx

```diff
@@ -14,72 +14,71 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import { Text, Box, ButtonSecondary, Link } from 'design';
+import React, { useEffect } from 'react';
+import {
+  Text,
+  Box,
+  ButtonSecondary,
+  Link,
+  Indicator,
+  ButtonLink,
+} from 'design';
 import { DialogContent, DialogFooter } from 'design/Dialog';
 import TextSelectCopy from 'teleport/components/TextSelectCopy';
 import DownloadLinks from 'teleport/components/DownloadLinks';
+import { State } from '../useAddApp';
 
 export default function Manually({
   isEnterprise,
   user,
   version,
   onClose,
   isAuthTypeLocal,
+  token,
+  createToken,
+  expires,
+  attempt,
 }: Props) {
   const { hostname, port } = window.document.location;
   const host = `${hostname}:${port || '443'}`;
   let tshLoginCmd = `tsh login --proxy=${host}`;
 
+  useEffect(() => {
+    createToken();
+  }, []);
+
```

Comment:
er actually, this causes a problem with our automatic tab b/c the token is shared between the two tabs. the auto tab will create a script with empty name and url (which users have to fill in prior to getting the script).

we should probably create the command after validation if token is already available

Reply:
what we can do is:

- create the token as before (move it back to the custom hook)
- in auto: after validation, create and render the command

Reply:
Hi Lisa, thanks for noticing this!

So this is what I did:
- moved useEffect back to hook
- added a new state variable `showCmd` to `Automatic` that start false and is set to true only when the form is valid
- only shows the script (and change the texts) when should `showCmd` is true (instead of when `cmd` exists, the way it used to be fone)

Commit: https://github.com/gravitational/webapps/pull/641/commits/fad2aa3e66e262f524b827b064d4772c09f4a150